winlogtimeline/collector/collect.py

```diff
@@ -1,42 +1,73 @@
-import pyevtx
-import winlogtimeline.util as util
+import Evtx.Evtx as evtx
+from threading import Thread
 
 
 # Note: It may be useful to take config out of this whole equation. Config could store the default filter configuration,
 # and that could be copied into the project config to allow user modifications.
 def import_log(log_file, project, config):
     """
-    Main routine to import an event log file into a project.
+    Main routine to import an event log file.
     :param log_file: A path to an event log file.
     :param project: A project instance.
     :param config: A config dictionary.
     :return: None
     """
-    # Open the event log
-    event_file = pyevtx.file()
-    event_file.open(log_file)
+    with evtx.Evtx(log_file) as log:  # Open the event logs.
+        evtx_records = collect_records(log)  # Get existing records.
+
+    return "Finished openning the event logs file."
+
     # Pull the logs from the event log file
-    event_records = collect_records(event_file) + collect_deleted_records(event_file)
-    machine_name = get_machine_name(event_file)
-    logs = (util.logs.parse_record(record, machine_name) for record in event_records)
-    logs = filter_logs(logs, project, config)
-    for log in logs:
-        project.write_log_data(log)
+    # event_records = collect_records(event_file) + collect_deleted_records(event_file)
+    # machine_name = get_machine_name(event_file)
+    # logs = (util.logs.parse_record(record, machine_name) for record in event_records)
+    # logs = filter_logs(logs, project, config)
+    # for log in logs:
+    #    project.write_log_data(log)
 
     # Pull verification data from the event log file
-    file_hash = get_log_file_hash(log_file)
-    project.write_verification_data(file_hash, log_file)
+    # file_hash = get_log_file_hash(log_file)
+    # project.write_verification_data(file_hash, log_file)
 
     # Close the event log
-    event_file.close()
+    # event_file.close()
+
+
+def get_xml_records(records):
+    """
+    Parse the given records into an XML format.
+    :param records: list of Evtx Record objects
+    :return: list of XML representations for record objects
+    """
+
+    xml_records = []  # 249.23 sec to complete. 31323 records - This was the fastest option I could achieve.
```

Comment:
Using xmltodict, I was able to get a list of OrderedDictionaries from 25425 records in 11.19 

Assuming the record file is in f, and that the relevant libraries are imported:
`start = time.time(); records = [xmltodict.parse(f.get_record(i).xml_string) for i in range(f.get_number_of_records())]; end = time.time()`

I could be missing something here, but I can't see what it is. I know you'll definitely save a ton of time by using the raw xml string instead of grabbing it from a function, but I don't know if the library change actually has much of an impact.

Reply:
Interesting. You seem to be using the "old" library (i.e. libevtx). The python-evtx library doesn't have those function calls. Even with xmltodict.parse, it takes a total of ~310 sec for 31323 records.

Please hold on this comment for now. I'm performing more tests on the subject.

Reply:
Yep, that would be my mistake. Less data means less time to parse I guess.